proxy/envoy/config.go

```diff
@@ -167,8 +167,15 @@ func buildHTTPListener(routeConfig *HTTPRouteConfig, ip string, port int, rds bo
 		config.RouteConfig = routeConfig
 	}
 
+	var sslContext *SSLContext
+	// Build the sslContext for http inbound listener (only inbound listener does not use rds)
+	if !rds && mesh.EnableAuth {
```

Comment:
The previous PR set rds to true for outbound http listener and false for inbound listener. I just follow the convention. 

Yes, the current logic uses auth without RDS.  

If rds check is not safe, we have two potential solutions 
- Add another bool to indicate whether it's a inbound listener or outbound
- Check whether the IP is wildcard. But that may not be safe either since an inbound listener may have a wildcard address (right?)